target/linux/ramips/dts/DWR-512-B.dts

```diff
@@ -0,0 +1,115 @@
+/dts-v1/;
+
+/include/ "rt5350.dtsi"
+
+/ {
+	compatible = "ralink,rt5350-soc";
+	model = "D-Link DWR-512 B";
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 0 1>;
+			linux,code = <0x211>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		sms {
+			label = "dwr-512-b:green:sms";
+			gpios = <&gpio0 8 1>;
+		};
+		status {
+			label = "dwr-512-b:green:status";
+			gpios = <&gpio0 9 1>;
+		};
+		2g {
+			label = "dwr-512-b:green:2g";
+			gpios = <&gpio0 17 1>;
+		};
+		3g {
+			label = "dwr-512-b:green:3g";
+			gpios = <&gpio0 19 1>;
+		};
+		sstrengthr {
+			label = "dwr-512-b:red:sigstrength";
+			gpios = <&gpio0 20 1>;
+		};
+		sstrengthg {
+			label = "dwr-512-b:green:sigstrength";
+			gpios = <&gpio0 21 1>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+ 
+		modem3g_enable {
+			gpio-export,name = "modem3g_enable";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 18 1>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	mx25l6405d@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "macronix,mx25l6405d", "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+		fast-read;
+
+                partition@0 {
+                        label = "u-boot";
+                        reg = <0x0 0x30000>;
+                        read-only;
+                };
+
+                partition@30000 {
+                        label = "u-boot-env";
+                        reg = <0x30000 0x10000>;
+                        read-only;
+                };
+
+                factory: partition@40000 {
+                        label = "factory";
+                        reg = <0x40000 0x10000>;
+                        read-only;
+                };
+
+                partition@50000 {
+                        label = "firmware";
+                        reg = <0x50000 0x7a0000>;
+                };
+
+		config: partition@7f0000 {
+			label = "config";
+			reg = <0x7f0000 0x10000>;
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "jtag", "uartf";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&config 0xe07a>;
```

Comment:
Dear Mathias,
I explain you the reason why I set the mtd-eeprom to 0xe07a.
I understood that the wmac in the dts point the memory location where the mac address is located.
In the capture file of the original firmware I found the wmac address:
`rt2860v2_ap: module license 'unspecified' taints kernel.
Length of MAC:6
mac:78:54:2e:a0:78:6e:`
Then I search in the two auxiliary flash partition (UI and config):
`Creating 6 MTD partitions on "raspi":`
`0x00000000-0x00800000 : "Whole"`
`0x00000000-0x00010000 : "Bootloader"`
`0x00010000-0x00150000 : "Kernel"`
`0x00150000-0x00710000 : "RootFS"`
`0x00710010-0x007f0000 : "UI"`
`mtd: partition "UI" doesn't start on an erase block boundary -- force read-only`
`0x007f0000-0x00800000 : "Config"`
 for this value and I found it in the config partition at address 0xe07e and I subtracted 0x4.
`root@lede:~# hexdump -v -s 0xe060 -n 64 -C /dev/mtd7`
`0000e060  80 0d 00 44 4c 4b 36 45  32 34 31 32 30 30 31 00  |...DLK6E2412001.|`
`0000e070  01 42 00 80 02 00 24 6e  30 42 00 80 06 00 78 54  |.B....$n0B....xT|`
`0000e080  2e a0 78 6d 00 f5 00 80  00 02 50 53 00 01 00 50  |..xm......PS...P|`
`0000e090  18 21 d5 28 ff ff ff ff  ff ff ff ff ff ff ff ff  |.!.(............|`
@0xe08a I can also found the 0x5053 referencing the SoC.

Proceeding in this way, as you can see, the lede system is able to get the same mac address:
[   12.321593] rt2800_wmac 10180000.wmac: loaded eeprom from mtd device "config"
[   12.335989] ieee80211 phy0: rt2x00_set_rt: Info - RT chipset 5350, rev 0500 detected
[   12.351537] ieee80211 phy0: rt2x00_set_rf: Info - RF chipset 5350 detected
[   12.366677] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'

root@lede:~# ifconfig
wlan0     Link encap:Ethernet  HWaddr 78:54:2E:A0:78:6D
          inet6 addr: fe80::7a54:2eff:fea0:786d/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2311 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1197 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:490305 (478.8 KiB)  TX bytes:193370 (188.8 KiB)


PS: How can you properly format the hexdump in this git conversation?

Bye.